pyccel/stdlib/cwrapper/cwrapper.c

```diff
@@ -0,0 +1,216 @@
+#include "cwrapper.h"
+
+/*                                                              */
+/*                        CAST_FUNCTIONS                        */
+/*                                                              */
+
+/*
+** All functions listed down are based on C/python api
+** with more tolerance to different precisions.
+** Arguments : Python Object and  C Object as pointer
+** Return    : return false upon failure and raise Error
+** reference of the used c python api function:
+** https://docs.python.org/3/c-api/float.html#c.PyFloat_AsDouble
+** https://docs.python.org/3/c-api/complex.html#c.PyComplex_RealAsDouble
+** https://docs.python.org/3/c-api/complex.html#c.PyComplex_ImagAsDouble
+** https://docs.python.org/3/c-api/long.html#c.PyLong_AsLong
+** https://docs.python.org/3/c-api/long.html#c.PyLong_AsLongLong
+*/
+
+bool	PyComplex_to_Complex64(Pyobject *o, float complex *c)
+{
+	float	real_part;
+	float	imag_part;
+
+
+	real_part = (float)PyComplex_RealAsDouble(o);
+	imag_part = (float)PyComplex_ImagAsDouble(o);
+
+	*c = CMPLXF(real_part, imag_part);
+
+	return true;
+}
+
+bool	PyComplex_to_Complex128(Pyobject *o, double complex *c)
+{
+	double	real_part;
+	double	imag_part;
+
+	real_part = PyComplex_RealAsDouble(o);
+	imag_part = PyComplex_ImagAsDouble(o);
+
+	*c = CMPLXF(real_part, imag_part);
+
+	return true;
+}
+
+bool	PyInt64_to_Int64(PyObject *o, int64_t *i)
+{
+	long long	out;
+
+	out = PyLong_AsLongLong(o);
+	if (out == -1 && PyErr_Occurred())
+		return false;
+
+	*i = (int64_t)out;
+
+	return true;
+}
+
+bool	PyInt32_to_Int32(PyObject *o, int32_t *i)
+{
+	long	out;
+
+	out = PyLong_AsLong(o);
+	if (out == -1 && PyErr_Occurred())
+		return false;
+
+	*i = (int32_t)out;
+
+	return true;
+}
+
+bool	PyInt16_to_Int16(PyObject *o, int16_t *i)
+{
+	long	out;
+
+	out = PyLong_AsLong(o);
+	if (out == -1 && PyErr_Occurred())
+		return false;
+
+	*i = (int16_t)out;
+
+	return true;
+}
+
+bool	PyInt8_to_Int8(PyObject *o, int8_t *i)
+{
+	long	out;
+
+	out = PyLong_AsLong(o);
+	if (out == -1 && PyErr_Occurred())
+		return false;
+
+	*i = (int8_t)out;
+
+	return true;
+}
+
+bool	PyBool_to_Bool(Pyobject *o, bool *b)
+{
+	*b = o == PyTrue;
+
+	return true;
+}
+
+bool	PyFloat_to_Float(Pyobject *o, float *f)
+{
+	double	out;
+
+
+	out = PyFloat_AsDouble(o);
+	if (out  == -1.0 && PyErr_Occured())
+		return false;
+
+	*f = (float)out;
+	return true;
+}
+
+bool	PyDouble_to_Double(PyObject *o, double *d)
+{
+	double	out;
+
+	out = PyFloat_AsDouble(o);
+	if (out  == -1.0 && PyErr_Occured())
+		return false;
+
+	*d = out;
+
+	return true;
+}
+
+
+/*
+** A Cast function that convert numpy array variable into ndarray variable,
+** by copying its information and data to a new variable of type ndarray struct
+** and return this variable to be used inside c code.
+*/
+
+t_ndarray		PyArray_to_ndarray(PyObject *o)
+{
+	t_ndarray	array;
+
+	c.nd          = PyArray_NDIM(o);
```

Comment:
c = array ?

Reply:
yeah I finish up by keeping array declared inside function, I still don't know if i need to pass array by adress as argument

Reply:
It depends on how you plan to use the function. I think it's up to you. I like it as a return